pkg/client/client_test.go

```diff
@@ -1679,6 +1679,294 @@ func TestGetAllFeatureVariablesWithDecisionWithoutFeature(t *testing.T) {
+		varVariableMap[id] = entities.VariationVariable{
+			ID:    id,
+			Value: v.varVal,
+		}
+
+		variableMap[id] = entities.Variable{
+			DefaultValue: v.defaultVal,
+			ID:           id,
+			Key:          v.key,
+			Type:         v.varType,
+		}
+
+		mockConfig.On("GetVariableByKey", testFeatureKey, v.key).Return(v.varVal, nil)
+	}
+
+	testVariation := entities.Variation{
+		ID:             "22222",
+		Key:            "22222",
+		FeatureEnabled: true,
+		Variables:      varVariableMap,
+	}
+
+	testVariation.FeatureEnabled = true
+	testExperiment := entities.Experiment{
+		ID:         "111111",
+		Variations: map[string]entities.Variation{"22222": testVariation},
+	}
+	testFeature := getTestFeature(testFeatureKey, testExperiment)
+	testFeature.VariableMap = variableMap
+	mockConfig.On("GetFeatureByKey", testFeatureKey).Return(testFeature, nil)
+
+	mockConfigManager := new(MockProjectConfigManager)
+	mockConfigManager.On("GetConfig").Return(mockConfig, nil)
+
+	testDecisionContext := decision.FeatureDecisionContext{
+		Feature:       &testFeature,
+		ProjectConfig: mockConfig,
+	}
+
+	expectedFeatureDecision := getTestFeatureDecision(testExperiment, testVariation)
+	mockDecisionService := new(MockDecisionService)
+	mockDecisionService.On("GetFeatureDecision", testDecisionContext, testUserContext).Return(expectedFeatureDecision, nil)
+
+	notificationCenter := notification.NewNotificationCenter()
+	client := OptimizelyClient{
+		ConfigManager:      mockConfigManager,
+		DecisionService:    mockDecisionService,
+		logger:             logging.GetLogger("", ""),
+		notificationCenter: notificationCenter,
+	}
+	var numberOfCalls = 0
+	note := notification.DecisionNotification{}
+	callback := func(notification notification.DecisionNotification) {
+		note = notification
+		numberOfCalls++
+	}
+	mockDecisionService.On("OnDecision", mock.AnythingOfType("func(notification.DecisionNotification)")).Return(1, nil)
+	mockDecisionService.notificationCenter = notificationCenter
+	id, _ := mockDecisionService.OnDecision(callback)
+
+	assert.NotEqual(t, id, 0)
+	client.GetDetailedFeatureDecisionUnsafe(testFeatureKey, testUserContext, true)
+
+	decisionInfo := map[string]interface{}{"feature": map[string]interface{}{"featureEnabled": true, "featureKey": "test_feature_key", "source": decision.Source(""),
+		"sourceInfo": map[string]string{}, "variableValues": map[string]interface{}{"var_bool": true, "var_double": 2.0, "var_int": 20,
+			"var_json": map[string]interface{}{"field1": 12.0, "field2": "some_value"}, "var_str": "var"}}}
+	assert.Equal(t, numberOfCalls, 1)
+	assert.Equal(t, decisionInfo, note.DecisionInfo)
+}
+
+func TestGetDetailedFeatureDecisionUnsafeWithTrackingDisabled(t *testing.T) {
+	testFeatureKey := "test_feature_key"
+	testUserContext := entities.UserContext{ID: "test_user_1"}
+
+	type variable struct {
+		key        string
+		defaultVal string
+		varVal     string
+		varType    entities.VariableType
+		expected   interface{}
+	}
```

Comment:
The same struct type is declared in the previous test. I suggest declaring it only once inside this file.